zeebe/qa/update-tests/CACHE_README.md

```diff
@@ -0,0 +1,134 @@
+# Version Compatibility Matrix Caching
+
+The `VersionCompatibilityMatrix` uses a two-level caching strategy to minimize GitHub API calls:
+
+1. **Application-level cache**: The `CachedVersionProvider` decorator caches discovered versions (with metadata) to a local JSON file
+2. **CI-level cache**: GitHub Actions can cache this file across workflow runs for even faster execution
+
+## How It Works
+
+**First discovery** (cold start):
+- Fetches version tags from GitHub API
+- For each latest patch: checks if a release exists (additional API call)
+- Saves complete metadata (version, isReleased, isLatest) to `.cache/camunda-versions.json`
+
+**Subsequent runs** (warm cache):
+- Reads from cached file
+- Zero GitHub API calls
+- Instant version discovery
+
+## Cache Location
+
+The cache file is stored at: `.cache/camunda-versions.json` (relative to a given location, see below)
+
+**Path Resolution Strategy** (in priority order):
+1. **`GITHUB_WORKSPACE`**: Environment variable set in GitHub Actions workflows
+2. **`maven.multiModuleProjectDirectory`**: Maven property pointing to the multi-module project root
+- This is passed to the test JVM via `maven-failsafe-plugin` configuration in `pom.xml`
+3. **Fallback to `user.dir`**: If none of the above work (should never happen in normal scenarios)
```

Comment:
🔧 on linux workstations `$HOME/.cache` is used from all applications. I think we should put it under `$HOME/.cache/camunda-dev/` or something like that.
On MacOS I think it's `$HOME/.Cache` (not sure if it's equivalent since the FS might not distinguish between letter case).
For windows I have no idea.

Is there a cross platform way to get the user caching folder directly?

Reply:
I think the user dir is usually the one that the test is executed within (at least via maven) 🤔 not specifically the home directory.